src/swap/types.ts

```diff
@@ -67,18 +97,48 @@ export type SwapError = {
   error: string; // The error message
 };
 
+/**
+ * Note: exported as public Type
+ */
+export type SwapParams = GetQuoteParams | GetSwapParams;
+
+export type SwapAPIParams = GetQuoteAPIParams | GetSwapAPIParams;
+
+/**
+ * Note: exported as public Type
+ */
+export interface Transaction {
+  transaction: TransactionData;
+
+  withParams(params: TransactionParams): TransactionData;
+}
+
+/**
+ * Note: exported as public Type
+ */
+export type TransactionData = {
+  chainId: number; // The chain ID
+  data: `0x${string}`; // The data for the transaction
+  gas: bigint; // The gas limit
+  to: `0x${string}`; // The recipient address
+  value: bigint; // The value of the transaction
+  nonce?: number; // The nonce for the transaction
+  maxFeePerGas?: bigint | undefined; // The maximum fee per gas
+  maxPriorityFeePerGas?: bigint | undefined; // The maximum priority fee per gas
+};
```

Comment:
Should we document what each type is in the comment above the function?

Reply:
No next is fine, after all Types should feel easy to understand. Also a type works in pair on where it's used, so folks should be able to see where the Type is used.